### PR TITLE
feat/recently-active-section

### PR DESCRIPTION
**Pull request by TTomaszPWR**
I added widget RecentlyActive and I refactored theme.

### CODE REVIEW

File: lib/features/home_view/home_view.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import 'package:flutter/material.dart';
+
+class HomeView extends StatelessWidget {
+  const HomeView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return  Scaffold(
+      appBar: AppBar(
+        title: const Text('Home'),
```
Comment (simon-the-shark) [inline lib/features/home_view/home_view.dart:11]: I suppose this is a placeholder widget. Cause if it is, then cool (maybe some comment). If not, then this should be a localized string :)

File: lib/features/home_view/widgets/recetly_active_section.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          style: context.textTheme.headline,
+        ),
+        const SizedBox(height: 15),
+        Expanded(
+          child: ListView.separated(
+            itemCount: 5,
+            separatorBuilder: (context, index) => const SizedBox(height: 8),
+            itemBuilder: (context, index) {
+              return RecentlyActiveGameWidget(
+                gameName: 'Gra $index',
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/recetly_active_section.dart:24]: This `Gra` should be localized

File: lib/features/home_view/widgets/recetly_active_section.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          context.localize.recently_active,
+          style: context.textTheme.headline,
+        ),
+        const SizedBox(height: 15),
+        Expanded(
+          child: ListView.separated(
+            itemCount: 5,
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/recetly_active_section.dart:20]: I guess this is a temporary value as well, but still don't like it :P

Comment (simon-the-shark) [reply lib/features/home_view/widgets/recetly_active_section.dart:20]: but you can leave it i guess. Or maybe extract it to local variable?

File: lib/features/home_view/widgets/recetly_active_section.dart [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+class RecentlyActiveGameWidget extends StatelessWidget {
+  const RecentlyActiveGameWidget({
+    super.key, 
+    required this.gameName,
+    required this.onTap
+  });
+
+  final String gameName;
+  final Function() onTap;
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/recetly_active_section.dart:44]: I encourage to always use full functional types with return type as well:

```dart
 final void Function() onTap;
```

I believe for this one, there's a shortcut alias type:
```dart
final VoidCallback onTap;
```

Comment (simon-the-shark) [reply lib/features/home_view/widgets/recetly_active_section.dart:44]: also this callback can probably be nullable as it doesn't make us any difference to accept null as well?

File: lib/features/home_view/widgets/recetly_active_section.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+              ),
+            ),
+            Container(
+              padding: const EdgeInsets.all(8),
+              decoration: BoxDecoration(
+                color: context.colorTheme.grey.withOpacity(0.13),
+                borderRadius: const BorderRadius.only(
+                  topRight: Radius.circular(7),
+                  bottomRight: Radius.circular(7),
+                ),
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/recetly_active_section.dart:73]: This  specific `Radius.circular(7)` is repeated multiple times, so we can probably extarct it to some config file or smth.

Let's say smth like this in `lib\config.dart`
```dart
abstract class RecentlyActiveConfig {
static const radius = Radius.circular(7); // if you have some better names then use them
}
```

File: lib/features/home_view/widgets/recetly_active_section.dart [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(8),
+              child: Text(
+                gameName,
+                style: context.textTheme.smallBoldText,
+              ),
+            ),
+            Container(
+              padding: const EdgeInsets.all(8),
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/recetly_active_section.dart:67]: this specific padding is only used 2 times, but still you can consider putting it in config I guess

File: lib/main.dart [hunk 7]
```diff
@@ -15,12 +18,16 @@ class MyApp extends StatelessWidget {
...
+        extensions: const [AppTheme()],
+        colorScheme: const ColorScheme.light().copyWith(
+            surface: Colors.white,
+          ),
       ),
+      debugShowCheckedModeBanner: false,
       localizationsDelegates: AppLocalizations.localizationsDelegates,
       supportedLocales: AppLocalizations.supportedLocales,
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+      //home: const MyHomePage(title: 'Flutter Demo Home Page'),
```
Comment (simon-the-shark) [inline lib/main.dart:29]: what this comment is doing out here. Is this some legacy trash code or will it be usefull?

File: lib/theme/app_color_theme.dart [hunk 8]
```diff
@@ -4,14 +4,15 @@ import 'package:theme_tailor_annotation/theme_tailor_annotation.dart';
 
 part 'app_color_theme.tailor.dart';
 
-@tailorMixinComponent
+@TailorMixinComponent()
```
Comment (simon-the-shark) [inline lib/theme/app_color_theme.dart:7]: just curious, why this change :P ??

**Review by simon-the-shark — COMMENTED** [review]
Good job. I haven't yet run your app, only left some notes from what I see here. Always happy do discuss. @mikolaj-jalocha you can check it out as well and tell us what you think, maybe you'll spot something interesting too

File: lib/features/home_view/widgets/recetly_active_section.dart (hunk 5 above)
Comment (simon-the-shark) [reply lib/features/home_view/widgets/recetly_active_section.dart:73]: We just don't like repeating hardcoded strings and ints. But also, let's not go to extremes.

File: lib/features/home_view/home_view.dart (hunk 1 above)
Comment (TTomaszPWR) [reply lib/features/home_view/home_view.dart:11]: yea it's placeholder I'll add comment

File: lib/theme/app_color_theme.dart (hunk 8 above)
Comment (TTomaszPWR) [reply lib/theme/app_color_theme.dart:7]: I was curious if there is a difference between them :)

File: lib/features/home_view/widgets/recetly_active_section.dart (hunk 5 above)
Comment (mikolaj-jalocha) [reply lib/features/home_view/widgets/recetly_active_section.dart:76]: here def. worth extracitng imo ;)

**Review by mikolaj-jalocha — COMMENTED** [review]
@simon-the-shark noted the most important changes here. Looks good now imo :-)

File: lib/theme/app_color_theme.dart (hunk 8 above)
Comment (simon-the-shark) [reply lib/theme/app_color_theme.dart:7]: i'm not an expert but the difference is that `@tailorMixinComponent` doesn't take any arguments and `@TailorMixinComponent( // can have some args specified here)` can specify some args with nondefault values. As i said, i've never educated myself on the topic but that's the difference in all these annotations i believe
